tools/convert_tests/convert_tests_helper.py

```diff
@@ -0,0 +1,409 @@
+from datetime import datetime
+import pytz
+import re
+
+# Define short names for various data types
+short_name_map = {
+    "string": "str",
+    "boolean": "bool",
+    "varbinary": "vbin",
+    "timestamp": "ts",
+    "timestamp_tz": "tstz",
+    "interval_year": "iyear",
+    "interval_day": "iday",
+    "interval": "iday",
+    "decimal": "dec",
+    "precision_timestamp": "pts",
+    "precision_timestamp_tz": "ptstz",
+    "fixedchar": "fchar",
+    "varchar": "vchar",
+    "fixedbinary": "fbin",
+}
+
+long_name_map = {v: k for k, v in short_name_map.items()}
+
+# Mapping of timezone abbreviations to pytz time zones
+timezone_abbr_map = {
+    "PST": "America/Los_Angeles",
+    "EST": "America/New_York",
+    "CST": "America/Chicago",
+    "MST": "America/Denver",
+    "HST": "Pacific/Honolulu",
+    "UTC": "UTC",
+}
+
+SQUOTE = "\u200B"
+DQUOTE = "&"
+
+# Map timezone abbreviations to valid pytz timezone names
+timediff_abbr_map = {
+    "-0800": "US/Pacific",  # Pacific Standard Time
+    "-0500": "US/Eastern",  # Eastern Standard Time
+}
+
+timediff_zone_map = {
+    "-0800": "PST",  # Pacific Standard Time
+    "-0500": "EST",  # Eastern Standard Time
+}
+
+
+def convert_type(type_str, mapping):
+    """Helper function to convert a type string using the provided mapping."""
+    if type_str.startswith("list<") and type_str.endswith(">"):
+        # Extract the inner type and convert it
+        inner_type = type_str[5:-1]  # Extract what's inside "list<...>"
+        short_inner_type = mapping.get(inner_type.lower(), inner_type)
+        return f"list<{short_inner_type}>"
+
+    base_type, parameters = (type_str.split("<", 1) + [""])[:2]
+    parameters = f"<{parameters}" if parameters else ""
+    return mapping.get(base_type.lower(), base_type) + parameters
+
+
+def convert_to_long_type(type_str):
+    type_str = convert_type(type_str, long_name_map)
+    if type_str == "interval_year" or type_str == "interval_day":
+        type_str = "interval"
+    return type_str
+
+
+def match_sql_duration(value, value_type):
+    """
+    Check if a string is in the format of 'X days, HH:MM:SS'.
+    Returns a match object if successful, or None if it doesn't match.
+    """
+    if value_type != "str":
```

Comment:
The docstring says it matches strings -- should the caller be making that determination.  If so, value_type isn't needed in the signature.